components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/AccountStorage.kt

```diff
@@ -36,9 +36,13 @@ internal class SharedPrefAccountStorage(
             // In case we switched from SecureAbove22AccountStorage to this implementation, migrate persisted account
             // and clear out the old storage layer.
             val secureStorage = SecureAbove22AccountStorage(context, crashReporter, migrateFromPlaintextStorage = false)
-            secureStorage.read()?.let { secureAccount ->
-                this.write(secureAccount.toJSONString())
-                secureStorage.clear()
+            try {
+                secureStorage.read()?.let { secureAccount ->
+                    this.write(secureAccount.toJSONString())
+                    secureStorage.clear()
+                }
+            } catch (e: Error) {
+                // Migrating the storage didn't work, but we tried
```

Comment:
log this, and (follow-up is fine), consider sending it to sentry so that we can keep track of these events in the wild.

Also, can you add an explanatory comment, similar to one in InsecurePreferencesImpl21 migration logic?